### PR TITLE
feat(refs T33577): Add new import tab

### PR DESCRIPTION
**Pull request by gruenbergerdemos**
**Ticket:** https://yaits.demos-deutschland.de/T33577

We need a new import tab for importing statements in xls formate from other projects. 
FE: Basically just copied `ExcelImport.vue` and changed route and permission and removed the radio buttons, because we can only import statements here. 

### PR Checklist

Delete the checkbox if it doesn't apply/isn't necessary.

- [ ] Tests updated/created
- [ ] Update documentation
- [ ] Link all relevant tickets
- [ ] Move the tickets on the board accordingly

### CODE REVIEW

**Comment by dresslerdemos** [comment]
Translation keys are to be added and required columns are not finalized, which results in (user presentable) import violations.

**Review by salisdemos — DISMISSED** [review]
looks ok for me

**Review by MoritzMandler — DISMISSED** [review]
lgtm - tested and works fine.

**Review by martendemos — CHANGES_REQUESTED** [review]

File: demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanStatementController.php [hunk 1]
```diff
@@ -2350,9 +2357,45 @@ public function readOnlyStatementListAction(
...
+        if (null === $procedure) {
+            throw ProcedureNotFoundException::createFromId($procedureId);
+        }
+
+        // recreate uploaded array
+        $uploads = explode(',', (string) $requestPost['uploadedFiles']);
+        $files = array_map([$fileService, 'getFileInfo'], $uploads);
+        $importer = $importerFactory->createXlsxStatementImporter($excelImporter);
+
+        return $this->importStatements($files, $procedureId, $importer);
```
Comment (martendemos) [inline demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanStatementController.php:2378]: This entire code part is duplicated. Is the reason the different permission or is it necessary to keep some kind of loose coupling for the addon system? I'm a bit confused.

File: demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanStatementController.php [hunk 2]
```diff
@@ -2364,43 +2407,31 @@ public function importStatementsAction(
...
-                    'confirm.statements.imported.from.xlsx',
-                    ['count' => $numberOfCreatedStatements, 'fileName' => $fileName]
-                );
-                $route = 'dplan_procedure_statement_list';
-                // Change redirect target if data input user
-                if ($permissions->hasPermission('feature_statement_data_input_orga')) {
-                    $route = 'DemosPlan_statement_orga_list';
-                }
+                $createdStatements = $importer->getCreatedStatements();
+                $numberOfCreatedStatements = is_countable($createdStatements) ? count($createdStatements) : 0;
```
Comment (martendemos) [inline demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanStatementController.php:2432]: getCreatedStatements() returns always an array. Is it even possible to be not countable? Is this check necessary?

File: demosplan/DemosPlanCoreBundle/Logic/Import/Statement/AbstractStatementSpreadsheetImporter.php [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+     * @var list<Statement>
+     */
+    protected array $generatedStatements = [];
+
+    /**
+     * {@link Tag} entities that do not yet exist in the database and were created during the import.
+     *
+     * @var list<Tag>
+     */
+    protected array $generatedTags = [];
```
Comment (martendemos) [inline demosplan/DemosPlanCoreBundle/Logic/Import/Statement/AbstractStatementSpreadsheetImporter.php:49]: Are the statements also not yet existing in the database and were created during import? Can we adjust the docblock of $generatedStatements and by bring it into line with the one of generatedTags?

File: demosplan/DemosPlanCoreBundle/Logic/Import/Statement/AbstractStatementSpreadsheetImporter.php [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    {
+        return $this->statementCopier->copyStatementObjectWithinProcedure(
+            $generatedOriginalStatement,
+            false,
+            true,
+            false
+        );
+    }
+
+    public function getErrorsAsArray(): array
```
Comment (martendemos) [inline demosplan/DemosPlanCoreBundle/Logic/Import/Statement/AbstractStatementSpreadsheetImporter.php:96]: missing documentation of definition of returned array

**Review by martendemos — COMMENTED** [review]
minors only

File: demosplan/DemosPlanCoreBundle/Logic/Import/Statement/AbstractStatementSpreadsheetImporter.php (hunk 4 above)
Comment (dresslerdemos) [reply demosplan/DemosPlanCoreBundle/Logic/Import/Statement/AbstractStatementSpreadsheetImporter.php:96]: The return is documented in the parent interface. However, I found some instances where the typehinting for the error array was actually missing.

File: demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanStatementController.php (hunk 1 above)
Comment (martendemos) [reply demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanStatementController.php]: minor only, could be better, but this isn't a blocker

File: demosplan/DemosPlanCoreBundle/Validator/ConsistentAnonymousOrgaConstraintValidator.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        if (null === $statement) {
+            return;
+        }
+
+        $orga = $statement->getMeta()->getOrgaName();
+        $department = $statement->getMeta()->getOrgaDepartmentName();
+        if ((User::ANONYMOUS_USER_ORGA_NAME === $orga && User::ANONYMOUS_USER_DEPARTMENT_NAME !== $department)
+            || (User::ANONYMOUS_USER_ORGA_NAME !== $orga && User::ANONYMOUS_USER_DEPARTMENT_NAME === $department)) {
+            $this->context->addViolation('If for a submitted statement either the organization or department is set as anonymous, both must be set in this way.'); // FIXME: add translation key
```
[the remark this answers is not in the document]
Comment (martendemos) [reply demosplan/DemosPlanCoreBundle/Validator/ConsistentAnonymousOrgaConstraintValidator.php]: she be done with 4363d992e0e8a54620f340369e9324b823083a6c

**Review by martendemos — APPROVED** [review]
lgtm

**Pull request merged**